### PR TITLE
Merge 'Linghe' branch to 'dev'

### PR DESCRIPTION
**Pull request by Linghe-Wang**
The 'dev' branch will be used for development purposes, while the 'main' branch will serve as the stable version. A simple code review will suffice. If you notice any bugs or have questions, feel free to follow up or send me a direct message.

### CODE REVIEW

File: flaskapp/app/App.py [hunk 1]
```diff
@@ -20,34 +25,45 @@
...
+        state = State([])
+        t = "This is what comes before: \"" + before + "\". Here is what comes after: \"" + after + "\". Please optimize this sentence: \"" + current + "\". The length of sentence should not be too long or too short than previous one."
+        #text_revise = "This is what comes before: \"" + before + "\". Here is what comes after: \"" + after + "\". Please optimize this paragraph: \"" + current + "\". The length of paragraph should not be too long or too short than previous one."
+        # if level:
+        #     t = text_revise
+        state.human_input = t
+        state = prompt(state)
+        return state.memory[-1][1]
 
-import pymongo
```
Comment (anmartin94) [inline flaskapp/app/App.py:24]: We are deleting this because we are no longer using Azure, correct?

File: extension/background.js [hunk 2]
```diff
@@ -26,13 +26,24 @@ chrome.runtime.onMessage.addListener(
         text = text.filter(function(element) {return element !== undefined;});
         projectID = request.project_id
         filename = request.editingFile;
+        if (request.message == "user_selection"){
```
Comment (anmartin94) [inline extension/background.js:31]: What does user_selection signify?

**Review by anmartin94 — APPROVED** [review]
Looks good.

File: extension/background.js (hunk 2 above)
Comment (Linghe-Wang) [reply extension/background.js:31]: "User selection" represents a user action which user selected a chunk of text and want Chatgpt help. When user doing this action, "user selection" message along with some other informations (selected text, context, line number, time) send to background.js.

File: flaskapp/app/App.py (hunk 1 above)
Comment (Linghe-Wang) [reply flaskapp/app/App.py:24]: We might need Ryan's confirmation.

**Pull request closed without merging**